base_import_xlsx/models/base_import.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2022 Therp BV <https://therp.nl>.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+"""Module overrides part of base_import to restore support for xlsx files."""
+import datetime
+import io
+
+from odoo import _, models
+from odoo.tools import DEFAULT_SERVER_DATE_FORMAT, DEFAULT_SERVER_DATETIME_FORMAT
+
+# pylint: disable=import-error
+from odoo.addons.base_import.models.base_import import FILE_TYPE_DICT
+
+try:
+    import openpyxl
+except ImportError:
+    openpyxl = None
+
+
+FILE_TYPE_DICT["application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"] = (
+    "xlsx",
+    openpyxl,
+    "openpyxl",
+)
+
+
+class BaseImport(models.TransientModel):
+    """Contains the method to read xlsx files."""
+
+    _inherit = "base_import.import"
+
+    def _read_xlsx(self, options):
+        """Read modern excel file using openpyxl.
+
+        Override method from super, that points back to _read_xls.
+        """
+        # pylint: disable=unused-argument
+        buffer = io.BytesIO(self.file or b"")
+        # read_only=False is needed because of a bug in openpyxl, combined with
+        # some providers of excel files not setting the correct dimensions.
+        book = openpyxl.load_workbook(
+            buffer, read_only=False, keep_vba=False, data_only=True, keep_links=False
+        )
+        sheet = book.worksheets[0]
+        for row in sheet.rows:
+            values = []
+            for cell in row:
+                if isinstance(cell, openpyxl.cell.read_only.EmptyCell):
+                    values.append("")
+                    continue
+                # Would like to skipp NULL values, but for some incomprehensible
+                # reason TYPE_NULL and TYPE_NUMERIC have the same value in openpyxl.
+                if cell.value is None:
+                    values.append("")
+                    continue
+                if cell.data_type == "d" and isinstance(cell.value, datetime.date):
```

Comment:
Is it intended that `timedelta` and `time` aren't handled? If I see correctly they will just appended in L82 and not even converted to string. I know it might be a weird case for export->import but if a customer is in between I could imagine it happening with the automatic typing of the calc tools.

Reply:
timedelta and time are not in the Odoo datatypes, so I do not quite see how this should be supported. I will convert unknown datatypes to string not in L82.

Reply:
Conversion to str would be handling it correctly I guess.